### PR TITLE
DDP-7072: FIx a bug when a user can register without entering diagnosis

### PR DESCRIPTION
**Pull request by alextanlan**
**_The idea_** - not validate composite question answers with null values.
And set null values explicitly if it is needed,
 e.g. in case when a user can enter/press the spacebar in the diagnosis field and it is accepted as input
(Pancan - Prequal)

### CODE REVIEW

**Review by MocanaAtBroad — COMMENTED** [review]
The fix seems to work/not allow submission of prequal if cancer is missing.
But there is a sequence that does not quite work as expected:
1. Go to page where you enter the cancer.
2. Press Next leaving the cancer empty. An error message is displayed
3. Select or type in a cancer
4. **The error message does not go away** (though you can click on next)

By contrast, compare to next page: enter age of 90000 and try to Submit.
You are blocked to proceed and error message for age is displayed. Now change age to 50. The age error is cleared once age is changed.

could you look into why the picklist error is not automatically cleared?

**Comment by alextanlan** [comment]
Marco, thanks for noticing! 

===

So, my thoughts about 
> The error message does not go away 
- it is another issue that we have in dev branch as well
- we used to skip a local validation for any questions before click on `Next`/`Prev` button in an activity. 
  Now we have a special case when we should force the local validation:
   an autocomplete picklist "cancer select" in Pancan study.
   
  Can you please take a look at 44548f204587a62fa412249e1c623e1a14bab12e.
  Seems like a crutch, but the case is out of our old validation logic, in my opinion.

**Review by MocanaAtBroad — DISMISSED** [review]

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/activityQuestion.component.ts [hunk 1]
```diff
@@ -76,12 +77,17 @@ export class ActivityQuestionComponent implements OnInit, OnDestroy {
             this.validationRequested$
         ]).pipe(
             // not displaying any local validations until a validation is requested
-            filter(([_, validationRequested]) => !!validationRequested),
+            // except some uniq cases
```
Comment (aglushko-broad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/activityQuestion.component.ts]: uniq -> unique

Comment (alextanlan) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/activityQuestion.component.ts]: fixed

**Pull request closed without merging**